presto-main/src/main/java/io/prestosql/operator/HashSemiJoinOperator.java

```diff
@@ -156,7 +167,14 @@ public void addInput(Page page)
                 }
             }
             else {
-                boolean contains = channelSet.contains(position, probeJoinPage);
+                boolean contains;
+                if (probeHashChannel.isPresent()) {
+                    long rawHash = BIGINT.getLong(page.getBlock(probeHashChannel.get()), position);
```

Comment:
Maybe hoist `page.getBlock(probeHashChannel.get()` outside of the `for` loop:
```
Optional<Block> hashBlock = probeHashChannel.map(page::getBlock);
for (...) {
...
    if (hashBlock.isPresent()) {
        long rawHash = BIGINT.getLong(hashBlock.get(), position);
        contains = channelSet.contains(position, probeJoinPage, rawHash);
    }
    else {
...
````